File: src/main/org/nlogo/workspace/ExtendableWorkspace.java

package org.nlogo.workspace;

// this is temporary to support a refactor of ExtensionManager

interface ExtendableWorkspace {
  public boolean compilerTestingMode();
  public String getSource(String filename) throws java.io.IOException;
  public boolean profilingEnabled();
  public org.nlogo.nvm.FileManager fileManager();
  public String attachModelDir(String filePath) throws java.net.MalformedURLException;
  public boolean warningMessage(String message);
  public Object readFromString(String path) throws org.nlogo.api.CompilerException;
}

File: netlogo-headless/src/main/workspace/ExtendableWorkspace.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.workspace

import org.nlogo.api.WorkspaceContext
import org.nlogo.nvm.{ FileManager, Tracer }

trait ExtendableWorkspace {
  def setProfilingTracer(tracer: Tracer): Unit
  def compilerTestingMode: Boolean
  def getSource(filename: String): String
  def profilingEnabled: Boolean
  def fileManager: FileManager
  def attachModelDir(filePath: String): String
  def warningMessage(message: String): Boolean
  def readFromString(path: String): AnyRef
  def workspaceContext: WorkspaceContext
}
